packages/fxa-auth-server/lib/payments/stripe-firestore.ts

```diff
@@ -152,4 +177,32 @@ export class StripeFirestore extends StripeFirestoreBase {
     }
     return paymentMethodSnap.docs[0].ref.delete();
   }
+
+  /**
+   * Remove the specified customer document and all subcollections
+   */
+  async removeCustomerRecursive(uid: string) {
+    const bulkWriterResults: string[] = [];
+    const bulkWriterErrors: BulkWriterError[] = [];
+    const bulkWriter = this.firestore.bulkWriter();
+    bulkWriter.onWriteResult((doc) => bulkWriterResults.push(doc.path));
+    bulkWriter.onWriteError((error) => {
+      if (error.failedAttempts < this.MAX_RETRY_ATTEMPTS) {
+        return true;
+      } else {
+        bulkWriterErrors.push(error);
+        return false;
+      }
+    });
+    try {
+      await this.firestore.recursiveDelete(
+        this.customerCollectionDbRef.doc(uid),
+        bulkWriter
+      );
```

Comment:
Neat, I didn't know they had that now!

Reply:
Same! It was a happy surprise finding it. Seems it's been there for a few years, but it's not widely known about yet, and the docs are still a little hidden away.

Reply:
Ditto! This is awesome!